session_5/Edurails/README.md

```diff
@@ -0,0 +1,180 @@
+# Associations and Validations
+
+## Tasks
+
+- A student has name, branch, admission year, e-mail and roll number.
+
+    In 'db/schema.rb' - 
+    ```ruby
+    create_table "students", force: :cascade do |t|
+        t.string "name"
+        t.string "branch"
+        t.string "admission_year", limit: 4
+        t.string "email"
+        t.string "roll_number"
+        t.datetime "created_at", precision: 6, null: false
+        t.datetime "updated_at", precision: 6, null: false
+    end
+    ```
+
+- A course has name, course code, branch, year, credits. 
+
+    In 'db/schema.rb' - 
+    ```ruby
+    create_table "courses", force: :cascade do |t|
+        t.string "name"
+        t.string "course_code"
+        t.string "branch"
+        t.string "year", limit: 4
+        t.integer "credits"
+        t.datetime "created_at", precision: 6, null: false
+        t.datetime "updated_at", precision: 6, null: false
+    end
+    ```
+
+
+- An assignment has name, submission deadline, weightage. 
+
+    In 'db/schema.rb' - 
+    ```ruby
+    create_table "assignments", force: :cascade do |t|
+    t.string "name"
+    t.datetime "deadline"
+    t.integer "weightage"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    end
+    ```
+
+- A student can register for many courses. 
+
+    In 'app/models/student.rb' -
+    ```ruby
+    has_many :courses
+    ```
+
+- A course can have many students. 
+
+    In 'app/models/course.rb' -
+    ```ruby
+    has_many :students
+    ```
+
+- A course has many assignments.
+
+    In 'app/models/course.rb' -
+    ```ruby
+    has_many :assignments
+    ```
+
+- A student can upload files for assignments for their registered courses.
+
+    In 'app/models/student.rb' -
+    ```ruby
+    has_many :assignments, through: :courses
+    ```
+
+- The only branches offered are `Computer Science and Engineering`,
+  `Mechanical Engineering`, `Mining Engineering` and `Electronics and
+  Communication Engineering`. 
+
+    In 'app/models/student.rb' -
+    ```ruby
+    validates :branch, inclusion: { :in => ["Computer Science and Engineering","Mechanical Engineering","Mining Engineering","Electronics and Communication Engineering"] }
+    ```
+    In 'app/models/course.rb' -
+    ```ruby
+    validates :branch, inclusion: { :in => ["Computer Science and Engineering","Mechanical Engineering","Mining Engineering","Electronics and Communication Engineering"] }
+    ```
+
+- The roll numbers follow same format as NITK: `#{Last two digits of admission year}#{Branch code}#{Three digit serial number}`.
+
+    In 'app/models/student.rb' -
+    ```ruby
+    validates :roll_number_validation
+    ```
+    ```ruby
+    def roll_number_validation
```

Comment:
Validation function names are of `{attribute_name}_(must|should|should_not)_(condition)`. For example, you can break this function into smaller functions:
- Validate first two numbers of  must match admission year, `roll_number_should_start_with_admission_year`.
- Validate branch code, `roll_number_must_have_valid_branch_code`.
- Validate serial number, `roll_number_must_end_with_serial_number`.

Reply:
Ohk..I'll change that.